net/api/impl/posix_server.h

```diff
@@ -190,6 +192,30 @@ class HTTPServerPOSIX final {
     handlers_.erase(path);
   }
 
+  struct StaticFileServer {
+    std::string body;
+    std::string content_type;
+    explicit StaticFileServer(const std::string& body, const std::string& content_type)
+        : body(body), content_type(content_type) {}
+    void operator()(Request r) { r.connection.SendHTTPResponse(body, HTTPResponseCode.OK, content_type); }
```

Comment:
I'll go with John's suggestion of GET only. It sounds natural to me that
POST-ing into static routes should result in 403 Method Not Allowed.

On Monday, February 23, 2015, Alex Zolotarev <notifications@github.com>
wrote:

> In net/api/impl/posix_server.h
> <https://github.com/KnowSheet/Bricks/pull/86#discussion_r25201074>:
>
> > @@ -190,6 +192,30 @@ class HTTPServerPOSIX final {
> >      handlers_.erase(path);
> >    }
> >
> > +  struct StaticFileServer {
> > +    std::string body;
> > +    std::string content_type;
> > +    explicit StaticFileServer(const std::string& body, const std::string& content_type)
> > +        : body(body), content_type(content_type) {}
> > +    void operator()(Request r) { r.connection.SendHTTPResponse(body, HTTPResponseCode.OK, content_type); }
>
> Why GET only? POST files also can serve static content if necessary.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KnowSheet/Bricks/pull/86/files#r25201074>.
>


-- 
Thanks,
Dima

http://dimakorolev.com/ | http://www.linkedin.com/in/dimakorolev |
dmitry.korolev@gmail.com | +1 (312) 593 2783